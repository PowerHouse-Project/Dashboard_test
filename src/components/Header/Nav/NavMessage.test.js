import React from "react";
import { render, screen } from "@testing-library/react";
import NavMessage from "./NavMessage";
import { messages1 } from "../../../assets/exports";

describe("NavMessage", () => {
  it("renders correctly", () => {
    const { asFragment } = render(<NavMessage />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("displays the correct number of messages", () => {
    render(<NavMessage />);
    expect(screen.getByText("You have 3 new messages")).toBeInTheDocument();
  });

  it("contains the correct message item", () => {
    render(<NavMessage />);
    const img = screen.getByAltText("Maria");
    expect(img).toHaveAttribute("src", messages1);
  });

  it("has the correct class names", () => {
    render(<NavMessage />);
    const listItems = screen.getAllByRole("listitem");
    expect(listItems[0]).toHaveClass("nav-item");
    expect(screen.getByRole("list")).toHaveClass("dropdown-menu");
  });
});
