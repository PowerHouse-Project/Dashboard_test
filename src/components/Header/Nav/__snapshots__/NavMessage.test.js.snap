// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavMessage renders correctly 1`] = `
<DocumentFragment>
  <li
    class="nav-item dropdown"
  >
    <a
      class="nav-link nav-icon"
      data-bs-toggle="dropdown"
      href="#"
    >
      <i
        class="bi bi-people"
      />
    </a>
    <ul
      class="dropdown-menu dropdown-menu-end dropdown-menu-arrow messages"
    >
      <li
        class="dropdown-header"
      >
        You have 3 new messages
        <a
          href="#"
        >
          <span
            class="badge rounded-pill bg-main-light-blue-neutral p-2 ms-2"
          >
            View all
          </span>
        </a>
      </li>
      <li>
        <hr
          class="dropdown-divider"
        />
      </li>
      <li
        class="message-item pl-5 pr-5 pt-2"
      >
        <a
          href="#"
        >
          <img
            alt="Maria"
            class="rounded-full w-12 h-12"
            src="messages-1.jpg"
          />
        </a>
        <div
          class="flex flex-col flex-wrap"
        >
          <h4>
            Maria Hudson
          </h4>
          <p>
            A family member has been added to your Home
          </p>
          <p>
            30 min. ago
          </p>
        </div>
      </li>
      <li>
        <hr
          class="dropdown-divider"
        />
      </li>
      <li
        class="dropdown-footer"
      >
        <a
          href="#"
        >
          Show all notifications
        </a>
      </li>
    </ul>
  </li>
</DocumentFragment>
`;
